src/OpenAL/OpenToolkit.OpenAL/AL/AL.cs

```diff
@@ -1,531 +1,1544 @@
-﻿//
-// AL.cs
+﻿// AL.cs
 //
-// Copyright (C) 2019 OpenTK
+// Copyright (C) 2020 OpenTK
 //
 // This software may be modified and distributed under the terms
 // of the MIT license. See the LICENSE file for details.
 //
 
 using System;
-using AdvancedDLSupport;
-using OpenToolkit.Core.Extensions;
-using OpenToolkit.Core.Loader;
+using System.Runtime.InteropServices;
+using System.Security;
+using OpenToolkit.Audio.OpenAL;
+using OpenToolkit.Core.Native;
 using OpenToolkit.Mathematics;
-using OpenToolkit.OpenAL.Attributes;
-using OpenToolkit.OpenAL.Interfaces;
-
-namespace OpenToolkit.OpenAL
+using OpenToolkit.OpenAL;
+
+/* Type Mapping
+// 8-bit boolean
+typedef char ALboolean;
+ * bool
+// character
+typedef char ALchar;
+ * byte
+// signed 8-bit 2's complement integer
+typedef char ALbyte;
+ * byte
+
+// unsigned 8-bit integer
+typedef unsigned char ALubyte;
+ * byte
+
+// signed 16-bit 2's complement integer
+typedef short ALshort;
+ * short
+
+// unsigned 16-bit integer
+typedef unsigned short ALushort;
+ * ushort
+
+// unsigned 32-bit integer
+typedef unsigned int ALuint;
+ * uint
+
+// signed 32-bit 2's complement integer
+typedef int ALint;
+ * int
+// non-negative 32-bit binary integer size
+typedef int ALsizei;
+ * int
+// enumerated 32-bit value
+typedef int ALenum;
+ * int
+
+// 32-bit IEEE754 floating-point
+typedef float ALfloat;
+ * float
+
+// 64-bit IEEE754 floating-point
+typedef double ALdouble;
+ * double
+
+// void type (for opaque pointers only)
+typedef void ALvoid;
+ * void
+*/
+
+namespace OpenToolkit.Audio.OpenAL
 {
     /// <summary>
-    /// Provides access to the OpenAL 1.1 API.
+    /// Provides access to the OpenAL 1.1 flat API.
     /// </summary>
-    public abstract class AL : NativeLibraryBase, IAL
+    [Api(Lib, typeof(OpenALLibraryNameContainer))]
+    public partial class AL : ApiContainer<AL>
     {
-        /// <summary>
-        /// Gets an instance of the API.
-        /// </summary>
-        /// <returns>The instance.</returns>
-        public static AL GetAPI()
+        internal const string Lib = "AL";
+        private const CallingConvention Style = CallingConvention.Cdecl;
+
+        static AL()
```

Comment:
Hmm, can static classes be partial? It's been a while since I had to try that. But if yes, then you can remove the private ctor and just make the class static, since its the container for the DllImports anyway, right?

Reply:
A static class can't inherit from other classes to `: ApiContainer<AL>` wouldn't work if `AL` was static

Reply:
You are absolutely right. I tried it out but forget to remove the comment :3 Sorry.